fail2ban/tests/sockettestcase.py

```diff
@@ -83,3 +90,17 @@ def testSocketForce(self):
 		self.server.stop()
 		serverThread.join(1)
 		self.assertFalse(os.path.exists(self.sock_name))
+
+
+class ClientMisc(unittest.TestCase):
+
+	def testPrintFormattedAndWiki(self):
+		# redirect stdout to devnull
+		saved_stdout = sys.stdout
+		sys.stdout = open(os.devnull, 'w')
+		try:
+			protocol.printFormatted()
+			protocol.printWiki()
+		finally:
+			# restore stdout
+			sys.stdout = saved_stdout
```

Comment:
Agree, your example looks very interesting also.
But here:
- it is not really needed (still not, because I'll rather cover a real f2b-client code, that ATM almost completely missing a coverage);
- something like that https://github.com/fail2ban/fail2ban/blob/master/fail2ban/tests/servertestcase.py#L930, as temporary replacement to `print` or channel `stdout` would be nice also, sufficient enough and much easier and clearer: 
I mean such "wrapper" like:
```python
	lines = []
	sys.stdout = AnyMemChannelTestClass(lambda *args: lines.append(args))
	...
	self.assertEqual(lines, testLines):
```